qiskit/pulse/instructions/delay.py

```diff
@@ -45,12 +45,11 @@ def __init__(self, duration: int,
             channel: The channel that will have the delay.
             name: Name of the delay for display purposes.
         """
-        self._channel = channel
         super().__init__((duration, channel), duration, (channel,), name=name)
 
     @property
     def channel(self) -> Channel:
         """Return the :py:class:`~qiskit.pulse.channels.Channel` that this instruction is
         scheduled on.
         """
-        return self._channel
+        return self.operands[1]
```

Comment:
This comment applies to this change to use of `operands` which is a little less clear than the original code. Could this be avoided with something like ParameterTable?

Reply:
Instruction arguments are an ordered list of operands and need not necessarily be named as their functionality is enforced by the data structure, and is not user-assignable. This is a pretty standard object model in intermediate representations, eg., [see](https://github.com/numba/llvmlite/blob/4acef2d5165768f43b315dd47ed99a5f8f63b3a8/llvmlite/ir/instructions.py#L18). This property just provides an easy way for the user to get a named value.

Reply:
I think the name of the property conveys the same information. The interface is just as clear, now I'm saying, "the first operand is the channel", rather than "the channel is the channel". I did look into using ParameterTable, but it didn't seem to be a good fit. In particular, here, this is necessary or else `operands[1]` and `_channel` aren't easily both updated upon assignment